File: app/src/main/java/qing/com/kotlin3/test/Main3Activity.java
package qing.com.kotlin3.test;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import qing.com.kotlin3.R;

public class Main3Activity extends AppCompatActivity {
    int arg = 1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main3);
        final ViewGroup viewGroup = (ViewGroup) findViewById(R.id.viewGroup);
        final TextView textView = new TextView(this);

        textView.setText("item" + (++arg));
        viewGroup.addView(textView);
        viewGroup.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(Main3Activity.this, "click="+textView.getText(), Toast.LENGTH_SHORT).show();
//                if (textView.getParent() != null) {
//                    ((ViewGroup) textView.getParent()).removeView(textView);
//                }
//                textView.setText("item" + (++arg));
//                viewGroup.addView(textView);
            }
        });
    }
}

File: lbsx64/src/main/java/lightplugin/posun/com/lbsx64/Main3Activity.kt
package lightplugin.posun.com.lbsx64

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main3.*

class Main3Activity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main3)
        click.setOnClickListener({view ->
            setResult(Activity.RESULT_OK, Intent().putExtra("name","tom"))
            finish()
        })
    }
}
